cdap-ui/app/cdap/components/KeyValuePairs/KeyValueStoreActions.js

```diff
@@ -44,8 +44,8 @@ const convertKeyValuePairsObjToMap = (keyValues) => {
 const keyValuePairsHaveMissingValues = (keyValues) => {
   if (keyValues.pairs) {
     return keyValues.pairs.some((keyValuePair) => {
-      if (keyValuePair.notDeletable && keyValuePair.provided) {
-        return false;
+      if (keyValuePair.notDeletable) {
+        return !keyValuePair.value;
```

Comment:
nit: Could we explicitly check for empty values instead of falsey values.

Reply:
done